lib/config.coffee

```diff
@@ -151,32 +153,9 @@ exports.validate = (config) ->
 	if disallowedProperty?
 		throw new Error("Validation: #{disallowedProperty} not recognized")
 
-###*
-# @summary Get a device configuration object from an application
-# @public
-# @function
-#
-# @param {String} application - application name
-# @param {Object} [options={}] - options
-# @param {String} [options.wifiSsid] - wifi ssid
-# @param {String} [options.wifiKey] - wifi key
-#
-# @returns {Promise<Object>} device configuration
-#
-# @todo Move this to the SDK
-#
-# @example
-# deviceConfig.getByApplication 'App1',
-# 	network: 'wifi'
-# 	wifiSsid: 'foobar'
-# 	wifiKey: 'hello'
-# .then (configuration) ->
-# 	console.log(configuration)
-###
-exports.getByApplication = (application, options = {}) ->
+getByApplication = (application, options = {}) ->
 	Promise.props
 		application: resin.models.application.get(application)
-		apiKey: resin.models.application.getApiKey(application)
```

Comment:
Rather than splitting into two methods with the same name, you could refactor these together using `deviceApiKey` in the options, and make this all simpler. This would become:


```
Promise.props
  ...
  apiKey: resin.models.application.getApiKey(application) if not options.deviceApiKey?
  ...
.then (results) ->
  ...

  if results.apiKey? 
    config.apiKey = results.apiKey
  else
    config.deviceApiKey = options.deviceApiKey

  exports.validate(config)
```

I think this then can subsume the entire `exports.getByApplication` method, and make `getByDevice` simpler too, as that can just take `deviceApiKey` in its options and pass those through to a single method, without having to branch or understand any of this. Does that work?

It does mean you _could_ pass `deviceApiKey` as an option to `getByApplication`. I'm not sure whether that's semantically sensible, which would be a good argument against this. Getting a config with a deviceApiKey back rather than an apiKey in that case though (i.e. `getByApplication({ deviceApiKey: 'abcd' })`) seems sensible enough to me though.